G1ANT.Addon.Net/G1ANT.Addon.Net/Commands/MailMoveToCommand.cs

```diff
@@ -0,0 +1,87 @@
+﻿/**
+*    Copyright(C) G1ANT Ltd, All rights reserved
+*    Solution G1ANT.Addon, Project G1ANT.Addon.Net
+*    www.g1ant.com
+*
+*    Licensed under the G1ANT license.
+*    See License.txt file in the project root for full license information.
+*
+*/
+
+using System;
+using MailKit;
+using MailKit.Net.Imap;
+using G1ANT.Language;
+using System.Net;
+
+
+namespace G1ANT.Addon.Net
+{
+    [Command(Name = "mail.moveto", Tooltip = "This command moves selected message to another folder.")]
+    public class MailMoveToCommand : Command
+    {
+        public class Arguments : CommandArguments
+        {
+            [Argument(Required = true, Tooltip = "Host name")]
+            public TextStructure Host { get; set; }
+
+            [Argument(Required = true, Tooltip = "Port")]
+            public IntegerStructure Port { get; set; } = new IntegerStructure(993);
+
+            [Argument(Required = true, Tooltip = "Login of the inbox user")]
+            public TextStructure Login { get; set; }
+
+            [Argument(Required = true, Tooltip = "Password of the inbox user")]
+            public TextStructure Password { get; set; }
+
+            [Argument(Required = true, Tooltip = "Mail to move")]
+            public MailStructure MailToMove { get; set; }
+
+            [Argument(Required = false, Tooltip = "Name of the origin folder")]
+            public TextStructure Origin { get; set; } = new TextStructure(String.Empty);
```

Comment:
Why do we need to pass original folder? Can we retrieve this information from the email we want to move?